Add all options including JSDoc to `Options` type

Lms24 opened the pull request:
This adds all options + JSDoc from the webpack plugin to our `Options` type. 

Some things in this PR we can discuss before merging, e.g. what to do with the `process.env` thing or if we can get rid of the `configFile` option (which I'd vote yes on). 

Others (e.g. `customHeaders` vs `customHeader`), I'd like to tackle in a follow-up PR as it involves modifying more code than just changing the type. 

ref: #62

lforst commented on packages/bundler-plugin-core/src/types.ts:
```diff
@@ -1,20 +1,61 @@
-  // dist: string,
+
+  /**
+   * 	Unique identifier for the distribution, used to further segment your release.
+   *  Usually your build number.
```
```suggestion
   * Unique identifier for the distribution, used to further segment your release.
   * Usually your build number.
```

Lms24 commented:
For future readers:

We decided to make slight deviations from the original Sentry Webpack plugin options:

* Removed `configFile` property as our new bundler plugins do not rely on Sentry CLI anymore. As a consequence, there's no need to specify a `.sentryclirc` file anymore. This means that parameters that were previously set in Sentry CLI which were also relevant for the Webpack plugin need to be set explicitly in the plugin options. 
* Removed implicit passing of plugin parameters via environment variables. Previously, it was possible to specify values implicitly as environment variables, such as `SENTRY_AUTH_TOKEN`. In our new plugin, you'll have to specify these values in the options. Note that this makes certain option fields **explicitly required** now which were previously only implicitly required.
  ```js
  // old config + environment variables were set for authToken, org and project 
  new SentryCliPlugin({
      include: './dist',
    })

  // new config (you can still use env variables but they need to be set explicitly):
  sentryWebpackPlugin({
    include: './dist',
    authToken: process.env.SENTRY_AUTH_TOKEN,
    org: process.env.SENTRY_ORG,
    project: process.env.SENTRY_PROJECT,
  });
  ```

Additional deviations:
* Removed `apiKey` property. This was never directly a part of the webpack plugin but the JS interface of Sentry CLI allowed for this property to be passed in. If you used only the webpack plugin before, this will most likely not affect you.

All of these changes will be written down in a migration guide coming up soon.
